### PR TITLE
Added field CVE to GitHub dependendatbot parser

### PR DESCRIPTION
**Pull request by ruedaja11**
I´ve added the field CVE to the github vulnerability parser because in SCA it is one of the most important field for a vulnerable component

### CODE REVIEW

**Comment by Maffooch** [comment]
Hi @ruedaja11 a couple things to do on this one before it cane be merged

1. Add unit tests. Add a redacted report with the CVE field. This is the file it would need to go into https://github.com/DefectDojo/django-DefectDojo/blob/master/unittests/tools/test_github_vulnerability_parser.py
2. Rebase against the bug fix branch so that this change can go into the release on Monday

**Comment by ruedaja11** [comment]
Hi @Maffooch Thank you for your feedback. I have added the unitest and I have change the branch against bugfix as you mentioned.

**Comment by Maffooch** [comment]
```
2023-01-12T17:28:25.6693204Z [36muwsgi_1         |[0m ======================================================================
2023-01-12T17:28:25.6693725Z [36muwsgi_1         |[0m FAIL: test_parse_cve (unittests.tools.test_github_vulnerability_parser.TestGithubVulnerabilityParser.test_parse_cve) (i=0)
2023-01-12T17:28:25.6694068Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2023-01-12T17:28:25.6694296Z [36muwsgi_1         |[0m Traceback (most recent call last):
2023-01-12T17:28:25.6694689Z [36muwsgi_1         |[0m   File "/app/unittests/tools/test_github_vulnerability_parser.py", line 256, in test_parse_cve
2023-01-12T17:28:25.6694980Z [36muwsgi_1         |[0m     self.assertEqual(finding.cve, "CVE-2021-42392")
2023-01-12T17:28:25.6695240Z [36muwsgi_1         |[0m AssertionError: None != 'CVE-2021-42392'
2023-01-12T17:28:25.6695389Z [36muwsgi_1         |[0m 
2023-01-12T17:28:25.6695731Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2023-01-12T17:28:25.6695948Z [36muwsgi_1         |[0m Ran 2245 tests in 572.622s
2023-01-12T17:28:25.6696096Z [36muwsgi_1         |[0m 
2023-01-12T17:28:25.6696322Z [36muwsgi_1         |[0m FAILED (failures=1, skipped=364)
```

**Comment by ruedaja11** [comment]
I think it is solved now! @Maffooch

**Comment by Maffooch** [comment]
```
2023-01-12T18:56:59.1681095Z [36muwsgi_1         |[0m ======================================================================
2023-01-12T18:56:59.1681698Z [36muwsgi_1         |[0m ERROR: test_parser_cve (unittests.tools.test_github_vulnerability_parser.TestGithubVulnerabilityParser.test_parser_cve) (i=0)
2023-01-12T18:56:59.1682135Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2023-01-12T18:56:59.1682464Z [36muwsgi_1         |[0m Traceback (most recent call last):
2023-01-12T18:56:59.1682949Z [36muwsgi_1         |[0m   File "/app/unittests/tools/test_github_vulnerability_parser.py", line 249, in test_parser_cve
2023-01-12T18:56:59.1683394Z [36muwsgi_1         |[0m     self.assertEqual(finding.cve_unsaved_vulnerability_ids[0], "CVE-2021-42392")
2023-01-12T18:56:59.1683737Z [36muwsgi_1         |[0m                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-01-12T18:56:59.1684187Z [36muwsgi_1         |[0m AttributeError: 'Finding' object has no attribute 'cve_unsaved_vulnerability_ids'
2023-01-12T18:56:59.1684421Z [36muwsgi_1         |[0m 
2023-01-12T18:56:59.1684787Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2023-01-12T18:56:59.1685100Z [36muwsgi_1         |[0m Ran 2246 tests in 581.057s
2023-01-12T18:56:59.1685276Z [36muwsgi_1         |[0m 
2023-01-12T18:56:59.1685511Z [36muwsgi_1         |[0m FAILED (errors=1, skipped=364)
```

Thi might be of help https://github.com/DefectDojo/django-DefectDojo/blob/master/readme-docs/DOCKER.md#run-the-tests-with-docker-compose

**Review by damiencarol — APPROVED** [review]

**Review by Maffooch — CHANGES_REQUESTED** [review]
Tests need to be passing before merge

File: unittests/tools/test_github_vulnerability_parser.py [hunk 1]
```diff
@@ -231,3 +231,26 @@ def test_parse_state(self):
...
+        for finding in findings:
+            finding.clean()
+
+        with self.subTest(i=0):
+            finding = findings[0]
+            self.assertEqual(finding.title, "RCE in H2 Console")
+            self.assertEqual(finding.severity, "Critical")
+            self.assertEqual(len(finding.unsaved_vulnerability_ids), 2)
+            self.assertEqual(finding.unsaved_vulnerability_ids[0], "GHSA-h376-j262-vhq6")
+            self.assertEqual(finding.cve_unsaved_vulnerability_ids[0], "CVE-2021-42392")
```
Comment (damiencarol) [inline unittests/tools/test_github_vulnerability_parser.py:249]: Please use `self.assertIn(` instead

**Review by damiencarol — CHANGES_REQUESTED** [review]

**Comment by Maffooch** [comment]
Closing due to inactivity

**Pull request closed without merging**